test/controllers/stages_controller_test.rb

```diff
@@ -396,4 +402,26 @@ def clone(method, format, extra = {})
       end
     end
   end
+
+  as_an_admin do
+    describe '#new' do
+      it 'can alter `does not deploy code`' do
+        get :new, params: {project_id: subject.project.to_param}
+        assert_select "#stage_no_code_deployed" do |input|
+          refute input.attr("disabled").present?
+        end
+      end
+    end
+
+    describe '#create' do
+      subject { assigns(:stage) }
+      it 'permits `no_code_deployed` as params' do
+        params = {project_id: projects(:test).to_param, stage: {name: 'test', no_code_deployed: true}}
+        post :create, params: params
+        subject.reload
+        subject.persisted?.must_equal(true)
```

Comment:
anything that can be reloaded is persisted ... so this test does not make much sense ?